### ISSUE TITLE
Refactor CLI classes so everything is environment aware

### ISSUE THREAD

**Issue by pugnascotia** [elastic/elasticsearch#49858]
After #49612 is merged, all CLI tools will support `-E` options (whether or not they do anything with them). We should look at the CLI class hierarchy and see if we can simplify it e.g. remove any unnecesary abstractions.

**Comment by williamrandolph** [comment]
@pugnascotia Is this still something you'd like to do? Otherwise I am not sure if this refactoring is enough of a priority to keep an open issue for it.

**Comment by pugnascotia** [comment]
Eh, it's very low priority.
